week2/ex1_done.py

```diff
@@ -0,0 +1,39 @@
+# 1.	Write a function that will parse a url query string like "page=2&foo=bar&x=y&utm_source=ss&utm_content=ccc" and return a query string where all the params are sorted by their names like this "foo=bar&page=2&x=y" and all the "utm_" parameters are removed.
+
+
+def parse_url(url):
+    string_splitted = url.split("&")
+    params_dict = {}
+    for entry in string_splitted:
+        key = value = ""
+        try:
+            key = entry.split("=")[0]
+            value = entry.split("=")[1]
+        except IndexError:
+            pass
+        if "utm_" not in key:
+            params_dict[key] = value
+    keys_sorted = sorted(params_dict.keys())
+    new_url = ""
+
+    for key in keys_sorted:
+
+        new_url += "{}={}{}".format(key, params_dict[key], "&")
```

Comment:
Don't use `+=` or `+` to build strings as it is very inefficient. Strings in python a immutable, and each concatenation creates new string. And also it hurts readability. 

Please use `["".join()](https://docs.python.org/3/library/stdtypes.html#str.join)` instead.